### ISSUE TITLE
Is it possible to enable text prompt for sam2 by adding text embedding model?

### ISSUE THREAD

**Issue by thesby** [facebookresearch/segment-anything-2#212]
`Grounding Dino + Sam2` is a popular way to segment object with text, but the detection module `Grounding Dino` may miss some object. So I think if I can add a text embedding model, such as T5, Sam2 maybe can segment by text prompt end-to-end. Is it possible to achieve this?

**Comment by HydraBucket** [comment]
This would be awesome. Id be keen to see this, too.

**Comment by GXcells** [comment]
Yes or also some way to fine tune it? (for example on specific use i microscopy images)

**Comment by JasonShen-SH** [comment]
Any ideas guys? This would definitely be awesome!
